ouroboros-consensus/src/Ouroboros/Consensus/NodeKernel.hs

```diff
@@ -538,11 +541,22 @@ forkBlockProduction maxBlockSizeOverride IS{..} BlockProduction{..} =
                -- We allow forging a block that is the successor of an EBB in
                -- the same slot.
              -> Right (headerPoint hdr, blockNo hdr)
-             | _ :> hdr' <- c'
-             -> Right (headerPoint hdr', blockNo hdr')
              | otherwise
-               -- If there is no block before it, so use genesis.
-             -> Right (genesisPoint, genesisBlockNo)
+             -> Right $ case c' of
+               _ :> hdr' -> (headerPoint hdr', blockNo hdr')
+               Empty p   -> mkPredecessor p (blockNo hdr)
+
+    mkPredecessor
+      :: Point (Header blk)
+         -- ^ point of predecessor of block @b@
+      -> BlockNo
+         -- ^ block number of @b@ (NOTE not of the predecessor)
+      -> (Point blk, BlockNo)
+         -- ^ point and block number of predecessor of @b@
+    mkPredecessor prevP bno
+        -- There is no block before it, so use genesis.
+      | 0 == bno  = (genesisPoint, genesisBlockNo)
+      | otherwise = (castPoint prevP, pred bno)
```

Comment:
Gah EBBs!

If @b@ (as defined by the Haddock on `mkPredecessor`) is an EBB, then `pred bno` is wrong; it should be `bno` in that case.